apps/publisher/extensions/assets/default/asset.js

```diff
@@ -313,8 +313,9 @@ asset.renderer = function(ctx) {
                 navList.push('Lifecycle', 'btn-lifecycle', util.buildUrl('lifecycle') + '/' + id);
             }
         }
-        if (permissionAPI.hasActionPermissionforPath(path, 'write', ctx.session) && permissionAPI.hasAssetPagePermission(type,'update',user.tenantId,username)) {
-           navList.push('Version', 'btn-copy', util.buildUrl('copy') + '/' + id);
+        if (permissionAPI.hasActionPermissionforPath(path, 'write', ctx.session)
```

Comment:
Thanks @splinter , Yes sure, it will improve the readability of the code. I have refactored it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/ddf074185b57470ae5dfcc110defc4eb4c4a40d5)